docs/source/cloudevents_spec.md

```diff
@@ -0,0 +1,274 @@
+# FlatBuffers Event Format for CloudEvents - Version 1.0.0-wip
+
+## Abstract
+
+The FlatBuffers Format for [CloudEvents](https://cloudevents.io) defines how events are expressed in the FlatBuffers Schema Language.
+
+## Table of Contents
+
+1. [Introduction](#1-introduction)
+2. [Attributes](#2-attributes)
+3. [Data](#3-data)
+4. [Transport](#4-transport)
+5. [Schema Evolution](#5-schema-evolution)
+6. [Examples](#6-examples)
+
+## 1. Introduction
+
+CloudEvents is a standardized and protocol-agnostic definition of the structure and metadata description of events. This specification defines how CloudEvents are to be represented using FlatBuffers.
+
+The Attributes section describes the naming conventions and data type mappings for CloudEvents attributes.
+
+This specification does not define an envelope format. The FlatBuffers type system's intent is primarily to provide a consistent type system for structured data serialization.
+
+The FlatBuffers event format does not currently define a batch mode format.
+
+### 1.1. Conformance
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC2119.
+
+## 2. Attributes
+
+This section defines how CloudEvents attributes are mapped to the FlatBuffers type system. This specification explicitly maps each attribute.
+
+### 2.1 Type System Mapping
+
+The CloudEvents type system MUST be mapped to FlatBuffers types as follows:
+
+| CloudEvents    | FlatBuffers                              |
+|----------------|------------------------------------------|
+| Boolean        | bool                                     |
+| Integer        | int                                      |
+| String         | string                                   |
+| Binary         | [ubyte] (vector of unsigned bytes)       |
+| URI            | string following RFC 3986 §4.3           |
+| URI-reference  | string following RFC 3986 §4.1           |
+| Timestamp      | string following RFC 3339 (ISO 8601)     |
+
+Extension specifications MAY define secondary mapping rules for the values of attributes they define, but MUST also include the previously defined primary mapping.
+
+### 2.2 REQUIRED Attributes
+
+The following attributes are REQUIRED to be present in all CloudEvents:
+
+- **id**: string (required)
```

Comment:
the "(required)" on each line seems redundant since line 52 already says they're all REQUIRED.